File: 프로그래머스/2/42860. 조이스틱/조이스틱.py
def solution(name):
    answer = 0
    up_down = 0
    
    for c in name:
        up=ord(c) - ord('A')
        down= ord('Z') - ord(c)+1
        up_down += min(up, down)
        # print(up_down)
        
    idx = 0
    left_right = len(name)-1

        
    for i in range(len(name)):
        next_idx = i+1
        while  next_idx<len(name) and name[next_idx] =='A':
            next_idx+=1
                    
        left_back= (len(name)-next_idx)*2 + i
        right_back = i + i + len(name)-next_idx
        print(left_back, right_back, left_right)
        left_right = min(left_back, right_back, left_right)
    
    return up_down + left_right

File: 프로그래머스/2/42860. 조이스틱/조이스틱.cpp
#include <string>
#include <vector>

using namespace std;

int solution(string name) {
    int answer = 0;
    
    for(auto n:name){
        int up = n-'A';
        int down = 'Z'-n+1;
        answer+=min(up, down);
    }
    int move = name.size()-1;
    
    for(int i=0; i<name.size(); i++){
        int next = i+1;
        while(next<name.size() && name[next]=='A'){
            next++;
        }
        int left_back = (name.size()-next)*2 + i;
        int right_back = i+i+name.size()-next;
        move = min(move, min(left_back, right_back));
    }
    answer+=move;
    return answer;
}